core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -102,7 +101,13 @@ class CityInfo {
         location = cityLocation
         setTransients()
 
-        setNewCityName(civInfo)
+        name = generateNewCityName(
+            civInfo.nation.cities,
+            civInfo.gameInfo.civilizations.flatMap { civilization ->
+                civilization.cities.map { city -> city.name }
```

Comment:
Does this mean that if I rename my cities, the same name will be chosen again? E.g., I found the city of Rome, then rename it to 'My Favourite City', and then found another city. To me it looks like Rome is then once again a valid choice, and as it is the first unused name, will be chosen again?

Reply:
Yes, and that's how it worked in the original game if I recall correctly.

Reply:
I'll check that as that sounds unintuitive.

Reply:
I think that's a good thing actually.

Reply:
> Yes, and that's how it worked in the original game if I recall correctly.

Turns out you're right

Reply:
Lol I remember that because I once renamed my capital while playing the original game only to find out my next newly-founded city will be named as the very name I've replaced. Maybe we can make an exception for capital names (i.e. the first element of `nationCityNames`), but this should be conferred with more people in my opinion.

Reply:
No, for that I'd vote for a persisted nextCityNameIndex. Yes the Huns could (should be able to?) steal the name right next turn, and that I'd find quite nice indeed.